tests/tool_fileredact_jpg_exif_stripping_test.php

```diff
@@ -0,0 +1,81 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_fileredact;
+
+use tool_fileredact\local\jpg;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/compatibility_trait.php');
+
+/**
+ * Defines names of plugin types and some strings used at the plugin managment
+ *
+ * @package   tool_fileredact
+ * @covers    \tool_fileredact\classes\local\jpg\strip_exif
+ * @author    Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright Catalyst IT, 2022
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class tool_fileredact_jpg_exif_stripping_test extends \advanced_testcase {
+    use compatibility_trait;
+
+    /**
+     * Check and ensure the fixture file has the markers we're testing for.
+     */
+    public function test_fixture_has_gps_markers() {
+        $pathname = __DIR__ . '/fixtures/gps.jpg';
+
+        $exiftoolexec = \escapeshellarg('exiftool');
+        exec("$exiftoolexec $pathname", $output);
+        $output = implode($output);
+        $this->compatible_assertStringContainsString('GPS Latitude', $output);
+        $this->compatible_assertStringContainsString('GPS Longitude', $output);
+    }
+
+    /**
+     * Test to ensure the JPG is processed, and returns no GPS information
+     */
+    public function test_method_removes_gps_markers() {
+        $pathname = $this->get_copy_of(__DIR__ . '/fixtures/gps.jpg');
+
+        // Convert example jpg.
+        (new jpg\strip_exif)->run((object) ['filename' => 'test.jpg'], ['pathname' => $pathname]);
+
+        // Ensure the final jpg does NOT contain javascript markers, thus indicating it was flattened.
+        $exiftoolexec = \escapeshellarg('exiftool');
+        exec("$exiftoolexec $pathname", $output);
```

Comment:
Can we do this same check in the exif strip method in realtime, and confirm nothing comes back?

Reply:
Yes, there's a bit of redundancy and overhead of another call but that can also be added in. 

Also I noticed in the referenced comment, the description is slightly wrong. There are no javascript markers in the EXIF data, and I'll change it to use a more relevant example.